tests/realtime/realtime.go

```diff
@@ -0,0 +1,211 @@
+package realtime
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+const tuneAdminRealtimeCloudInitData = `password: fedora
+chpasswd: { expire: False }
+bootcmd:
+  - sudo tuned-adm profile realtime
+`
+
+func setkernelSchedulerRealtimeRuntime(virtCli kubecli.KubevirtClient, nodeName, value string) error {
+	command := fmt.Sprintf("sysctl kernel.sched_rt_runtime_us=%s", value)
+	pod := tests.RenderPrivilegedPod("kernel-scheduler-realtime-runtime", []string{"/bin/bash"}, []string{"-c", command})
+	pod.Spec.NodeName = nodeName
+	pod.Spec.RestartPolicy = k8sv1.RestartPolicyOnFailure
+	_, err := virtCli.CoreV1().Pods(util.NamespaceTestDefault).Create(context.Background(), pod, k8smetav1.CreateOptions{})
+
+	return err
+}
+
+func byStartingTheVMI(vmi *v1.VirtualMachineInstance, virtClient kubecli.KubevirtClient) {
+	By("Starting a VirtualMachineInstance")
+	var err error
+	vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+	Expect(err).ToNot(HaveOccurred())
+	tests.WaitForVMIPodToBeScheduled(vmi, util.NamespaceTestDefault)
+	pod := tests.GetPodByVirtualMachineInstance(vmi)
+	setkernelSchedulerRealtimeRuntime(virtClient, pod.Spec.NodeName, "-1")
+	tests.WaitForSuccessfulVMIStart(vmi)
```

Comment:
I think this might be racy. Perhaps we should reschedule both the job and the VMI to the same node.
What if we'll just have a function similar to `SkipTestIfNoCPUManagerWith2MiHugepages` that will return a `nodeName` where we could schedule both the job and the vmi.
What do you think?

Reply:
The job will run in the same node where the VMI runs. The previous line will guarantee that the node where the job runs is the same where the pod is scheduled to run. 

The tricky part for this to work is to wait for the pod to be scheduled first to the node, and then run the job in that node while the pod is initializing. The job is quick enough that when the VMI pod starts, the node has already been set with the proper kernel parameter. But I can see that this might not always be the case if the image for the job fails to pull, for instance. I can evaluate that the job is completed before I wait for the VMI to start. 

Doing the scheduling job by ourselves is a bit complex, since knowing which node has huge pages available and then creating the VM is not an isolated action: we could end up with flaky cases where another pod is scheduled in the same node we plan to use and it consumes the available huge pages.